website/profile/views.py

```diff
@@ -83,22 +83,51 @@ def date_or_none(date):
         return None
 
 
+def validate_user(data, user):
+    """Check if the user in request is the user who log in """
+    if 'id' in data:
+        if data['id'] != user._id:
+            raise HTTPError(httplib.FORBIDDEN)
+    else:
+        # raise an error if request doesn't have user id
+        raise HTTPError(httplib.BAD_REQUEST, data={'message_long': '"id" is required'})
+
+@must_be_logged_in
+def resend_confirmation(auth):
+    user = auth.user
+    data = request.get_json()
+
+    validate_user(data, user)
+
+    try:
+        primary = data['email']['primary']
+        confirmed = data['email']['confirmed']
+        address = data['email']['address'].strip().lower()
+    except KeyError:
+        raise HTTPError(httplib.BAD_REQUEST)
+
+    if primary or confirmed:
+        raise HTTPError(httplib.BAD_REQUEST, data={'message_long': 'Cannnot resend confirmation for confirmed emails'})
+
+    user.add_unconfirmed_email(address)
+
+    # TODO: This setting is now named incorrectly.
```

Comment:
Oh, okay. I thought it was something you added in. @lyndsysimon, can you clarify what you mean in this TODO?